Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -1544,11 +1553,114 @@ def search_device_command():
+        endpoint = Common.Endpoint(
+            id=single_device.get('device_id'),
+            hostname=single_device.get('hostname'),
+            ip_address=single_device.get('local_ip'),
+            os=single_device.get('platform_name'),
+            os_version=single_device.get('os_version'),
+            status=status,
+            is_isolated=is_isolated,
+            mac_address=single_device.get('mac_address'),
+            vendor=INTEGRATION_NAME)
+
+        entry = get_trasnformed_dict(single_device, SEARCH_DEVICE_KEY_MAP)
+        headers = ['ID', 'Hostname', 'OS', 'MacAddress', 'LocalIP', 'ExternalIP', 'FirstSeen', 'LastSeen', 'Status']
+
+        command_results.append(CommandResults(
+            outputs_prefix='CrowdStrike.Device',
+            outputs_key_field='ID',
+            outputs=entry,
+            readable_output=tableToMarkdown('Devices', entry, headers=headers, headerTransform=pascalToSpace),
+            raw_response=raw_res,
+            indicator=endpoint
+        ))
+
+    return command_results
+
+
+def search_device_by_ip(raw_res, ip_address):
+    devices = raw_res.get('resources')
+    filtered_devices = []
+    for single_device in devices:
+        if single_device.get('local_ip') == ip_address:
+            filtered_devices.append(single_device)
+
+    if filtered_devices:
+        raw_res['resources'] = filtered_devices
+    else:
+        raw_res = None
+    return raw_res
+
+
+def generate_status_fields(endpoint_status):
+    status = ''
+    is_isolated = ''
+
+    if endpoint_status == 'normal':
+        status = 'Online'
+    elif endpoint_status == 'containment_pending':
+        is_isolated = 'Pending isolation'
+    elif endpoint_status == 'contained':
+        is_isolated = 'Yes'
+    elif endpoint_status == 'lift_containment_pending':
+        is_isolated = 'Pending unisolation'
+
+    return status, is_isolated
+
+
+def generate_endpoint_by_contex_standart(devices):
+    standart_endpoints = []
+    for single_device in devices:
+        status, is_isolated = generate_status_fields(single_device.get('status'))
+        endpoint = Common.Endpoint(
+            id=single_device.get('device_id'),
+            hostname=single_device.get('hostname'),
+            ip_address=single_device.get('local_ip'),
+            os=single_device.get('platform_name'),
+            os_version=single_device.get('os_version'),
+            status=status,
+            is_isolated=is_isolated,
+            mac_address=single_device.get('mac_address'),
+            vendor=INTEGRATION_NAME)
+        standart_endpoints.append(endpoint)
+    return standart_endpoints
+
+
+def get_endpoint_command():
+    args = demisto.args()
+    if 'id' in args.keys():
+        args['ids'] = args.get('id', '')
+
+    # handles the search by id or by hostname
+    raw_res = search_device()
+
+    if ip := args.get('ip'):
+        # there is no option to filter by ip, in this case we'll get all the devices
```

Comment:
if there is no ip/id - let's not perform a search to not crush the system - same as the bug with the AD query.